WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -215,24 +215,17 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
     }
 
     private func displayProductPreview() {
-        var permalink = URLComponents(string: product.permalink)
-        var updatedQueryItems = permalink?.queryItems ?? []
-        updatedQueryItems.append(.init(name: "preview", value: "true"))
-        permalink?.queryItems = updatedQueryItems
-        guard let url = permalink?.url else {
+        guard var permalink = URLComponents(string: product.permalink),
+              let nonce = ServiceLocator.stores.sessionManager.defaultSite?.frameNonce else {
             return
         }
 
-        let credentials = ServiceLocator.stores.sessionManager.defaultCredentials
-        guard let username = credentials?.username,
-              let token = credentials?.authToken,
-              let site = ServiceLocator.stores.sessionManager.defaultSite else {
-            return
-        }
+        var updatedQueryItems = permalink.queryItems ?? []
+        updatedQueryItems.append(.init(name: "preview", value: "true"))
+        updatedQueryItems.append(.init(name: "frame-nonce", value: nonce))
+        permalink.queryItems = updatedQueryItems
 
-        let configuration = WebViewControllerConfiguration(url: url)
-        configuration.secureInteraction = true
```

Comment:
Do you think we should keep `secureInteraction` enabled to disable the bottom toolbar? User is not expected to navigate far from the preview page, open it in Safari or share.

Reply:
Good point, I've added that back in [d947b78](https://github.com/woocommerce/woocommerce-ios/pull/8058/commits/d947b78ec31f3ffca9891b4a22755ada907839ff).